tests/test_transforms.py

```diff
@@ -258,3 +262,24 @@ def transform_class(cls):
                 import UserDict
         """
         )
+
+    def test_transform_aborted_if_recursion_limited(self):
+        def transform_call(node: Call) -> Const:
+            return node
+
+        self.transformer.register_transform(
+            nodes.Call, transform_call, _looks_like_dataclass_field_call
+        )
+
+        if IS_PYPY:
+            original_limit = 1000  # pypy doesn't expose this
+            sys.setrecursionlimit(600)
+        else:
+            original_limit = sys.getrecursionlimit()
+            sys.setrecursionlimit(1000)  # use the default
```

Comment:
```suggestion
        # pypy doesn't expose its recursion limit
        original_limit = 1000 if IS_PYPY else sys.getrecursionlimit()
        sys.setrecursionlimit(10)  # really low value to be able to test easily
```

Then we can use a smaller test fixture.

Reply:
10 is too low, it will cause other unrelated recursion errrors.

Reply:
but pypy seems like it does expose its recursion limit; i probably read outdated information while debugging. will update.